base/initdefs.jl

```diff
@@ -374,7 +375,7 @@ calls `exit(n)`, then Julia will exit with the exit code corresponding to the
 last called exit hook that calls `exit(n)`. (Because exit hooks are called in
 LIFO order, "last called" is equivalent to "first registered".)
 """
-atexit(f::Function) = (pushfirst!(atexit_hooks, f); nothing)
+atexit(f::Function) = Base.@lock _atexit_lock (pushfirst!(atexit_hooks, f); nothing)
```

Comment:
Does anyone have a **style preference** between this oneliner and this multiliner?:
```suggestion
function atexit(f::Function)
    Base.@lock _atexit_lock begin
        pushfirst!(atexit_hooks, f)
        return nothing
    end
end
```
I think I prefer the nested multiliner, but it's maybe less julian, so I want to leave it up to reviewers. Thanks!